### ISSUE TITLE
RuntimeError: This event loop is already running" if debugger is run from inside a running event loop

### ISSUE THREAD

**Issue by brianmaissy** [ipython/ipython#12140]
Since upgrading from ipython from 7.9.0 to 7.12.0, and from prompt_toolkit 2 to 3, if I start a debugger from inside a running event loop (via `ipdb.set_trace()`) I get the following exception:

```
../../.local/lib/python3.6/site-packages/IPython/terminal/debugger.py:114: in cmdloop
    line = self.pt_app.prompt()
../../.local/lib/python3.6/site-packages/prompt_toolkit/shortcuts/prompt.py:997: in prompt
    return self.app.run()
../../.local/lib/python3.6/site-packages/prompt_toolkit/application/application.py:810: in run
    return loop.run_until_complete(self.run_async(pre_run=pre_run))
/usr/lib/python3.6/asyncio/base_events.py:471: in run_until_complete
    self.run_forever()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <_UnixSelectorEventLoop running=False closed=False debug=False>

    def run_forever(self):
        """Run until stop() is called."""
        self._check_closed()
        if self.is_running():
>           raise RuntimeError('This event loop is already running')
E           RuntimeError: This event loop is already running

/usr/lib/python3.6/asyncio/base_events.py:425: RuntimeError
```

It looks like the prompt_toolkit app isn't meant to be run from inside an event loop. I don't know if this is considered an unsupported use case, or if it's a prompt_toolkit or an ipython problem.

possibly related:
* https://github.com/ipython/ipython/issues/12019
* https://github.com/ipython/ipython/pull/12028

**Comment by brianmaissy** [comment]
related issues in prompt_toolkit and ipdb:
* https://github.com/prompt-toolkit/python-prompt-toolkit/issues/1084
* https://github.com/gotcha/ipdb/issues/184

**Comment by jonathanslenders** [comment]
@brianmaissy : Can you have a look at this PR: https://github.com/ipython/ipython/pull/12141 and see whether that solves the issue?

I didn't test extensively, but I could reproduce your issue and this seems to fix it.

**Comment by brianmaissy** [comment]
Yeah looks great! solves the issue for me.

**Comment by Carreau** [comment]
Closed by #12141
